src/solvers/dg_structured/dg_3d.jl

```diff
@@ -0,0 +1,292 @@
+      for jj in eachnode(dg)
+        integral_contribution = derivative_dhat[jj, j] * flux2
+        add_to_node_vars!(du, integral_contribution, equations, dg, i, jj, k, element)
+      end
+
+      flux3 = transformed_flux(u_node, 3, mesh, equations)
+      for kk in eachnode(dg)
+        integral_contribution = derivative_dhat[kk, k] * flux3
+        add_to_node_vars!(du, integral_contribution, equations, dg, i, j, kk, element)
+      end
+    end
+  end
+
+  return nothing
+end
+
+
+function prolong2interfaces!(cache, u::AbstractArray{<:Any,5}, mesh::StructuredMesh, equations, dg::DG)
+  @threaded for element_ind in eachelement(dg, cache)
+    element = cache.elements[element_ind]
+
+    # Interfaces in x-direction
+    for k in eachnode(dg), j in eachnode(dg), v in eachvariable(equations)
+      element.interfaces[1].u_right[v, j, k] = u[v, 1, j, k, element_ind]
+      element.interfaces[2].u_left[v, j, k] = u[v, end, j, k, element_ind]
+    end
+
+    # Interfaces in y-direction
+    for k in eachnode(dg), i in eachnode(dg), v in eachvariable(equations)
+      element.interfaces[3].u_right[v, i, k] = u[v, i, 1, k, element_ind]
+      element.interfaces[4].u_left[v, i, k] = u[v, i, end, k, element_ind]
+    end
+
+    # Interfaces in z-direction
+    for j in eachnode(dg), i in eachnode(dg), v in eachvariable(equations)
+      element.interfaces[5].u_right[v, i, j] = u[v, i, j, 1, element_ind]
+      element.interfaces[6].u_left[v, i, j] = u[v, i, j, end, element_ind]
+    end
+  end
+
+  return nothing
+end
+
+
+function prolong2boundaries!(cache, u::AbstractArray{<:Any,5}, 
+    boundary_condition::BoundaryConditionPeriodic, mesh::StructuredMesh, equations, dg::DG)
+  @unpack size, linear_indices = mesh
+
+  # Boundaries in x-direction
+  for element_y in 1:size[2], element_z in 1:size[3]
+    @views cache.elements[1, element_y, element_z].interfaces[1].u_left .= u[:, end, :, :, linear_indices[end, element_y, element_z]]
+    @views cache.elements[end, element_y, element_z].interfaces[2].u_right .= u[:, 1, :, :, linear_indices[1, element_y, element_z]]
+  end
+
+  # Boundaries in y-direction
+  for element_x in 1:size[1], element_z in 1:size[3]
+    @views cache.elements[element_x, 1, element_z].interfaces[3].u_left .= u[:, :, end, :, linear_indices[element_x, end, element_z]]
+    @views cache.elements[element_x, end, element_z].interfaces[4].u_right .= u[:, :, 1, :, linear_indices[element_x, 1, element_z]]
+  end
+
+  # Boundaries in z-direction
+  for element_x in 1:size[1], element_y in 1:size[2]
+    @views cache.elements[element_x, element_y, 1].interfaces[5].u_left .= u[:, :, :, end, linear_indices[element_x, element_y, end]]
+    @views cache.elements[element_x, element_y, end].interfaces[6].u_right .= u[:, :, :, 1, linear_indices[element_x, element_y, 1]]
+  end
+
+  return nothing
+end
+
+
+function calc_interface_flux!(nonconservative_terms::Val{false}, mesh::StructuredMesh{3}, equations,
```

Comment:
I think you can just ignore these `nonconservative_terms` stuff right now and remove it from all your functions